### PR TITLE
Correction in datetime handling for crossword web publication date

### PR DESCRIPTION
**Pull request by shtukas**
This change makes the publishing datetime for crosswords
appear in UTC. In particular during summer time, when the
UK is at BST (UTC+1) the datetime appears as 2016-03-29T23:00:00.000+00:00

This fixes an existing issue by which crosswords are being published one hour late
during summer.

### CODE REVIEW

File: crossword-xml-uploader/src/main/scala/com/gu/crossword/crosswords/XmlProcessor.scala [hunk 1]
```diff
@@ -3,14 +3,14 @@ package com.gu.crossword.crosswords
...
 trait DateLogic {
 
   def transformDate(dateString: String): String = {
-    val inputFormat = DateTimeFormat.forPattern("dd.MM.yyyy HH:mm")
-    val outputFormat = ISODateTimeFormat.dateTime()
-    val isoFormattedOutput = LocalDateTime.parse(dateString, inputFormat).toDateTime(DateTimeZone.forID("Europe/London")).toString(outputFormat)
+    val inputFormat = DateTimeFormat.forPattern("dd.MM.yyyy HH:mm").withZone(DateTimeZone.forID("Europe/London"))
+    val outputFormat = ISODateTimeFormat.dateTime().withZone(DateTimeZone.forID("UTC"))
+    val isoFormattedOutput = DateTime.parse(dateString, inputFormat).toString(outputFormat)
     isoFormattedOutput.replaceAllLiterally("Z", "+00:00")
```
Comment (adamnfish) [inline crossword-xml-uploader/src/main/scala/com/gu/crossword/crosswords/XmlProcessor.scala]: I realise it's not part of this change, but doing a string replace on the output is a bit odd. Better to just use the right format in the first place.

This should work:
```scala
DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ssZZ")
```

see [Zone in this docs page](http://joda-time.sourceforge.net/apidocs/org/joda/time/format/DateTimeFormat.html)

Comment (LATaylor-guardian) [inline crossword-xml-uploader/src/main/scala/com/gu/crossword/crosswords/XmlProcessor.scala]: FYI replaceAllLiterally is an expensive operation as we found recently, see: https://github.com/guardian/content-api-models/commit/fcbbdbbc489e07dc9d3699f5428012fa25db8d6a

This doesn't matter in this use, but is worth being aware of.

Comment (shtukas) [inline crossword-xml-uploader/src/main/scala/com/gu/crossword/crosswords/XmlProcessor.scala]: Thanks Luke. I didn't give too much thoughts to it, but I was wondering why they had used it....

**Comment by philmcmahon** [comment]
👍 I can show you how to deploy this later.

**Comment by LATaylor-guardian** [comment]
@philmcmahon Probably worth making this deployable by Riff Raff at some point.
